opm/simulators/wells/MSWellHelpers.cpp

```diff
@@ -220,44 +220,31 @@ invDX(const MatrixType& D, VectorType x, DeferredLogger& deferred_logger)
 }
 
 template <typename ValueType>
-ValueType calculateFrictionFactor(const double area, const double diameter,
-                                  const ValueType& w, const double roughness,
-                                  const ValueType& mu)
+ValueType frictionPressureLoss(const double l, const double diameter,
+                               const double area, const double roughness,
+                               const ValueType& density,
+                               const ValueType& w, const ValueType& mu)
 {
-
-    ValueType f = 0.;
     // Reynolds number
     const ValueType re = abs( diameter * w / (area * mu));
 
-    if ( re == 0.0 ) {
-        // make sure it is because the mass rate is zero
-        assert(w == 0.);
-        return 0.0;
-    }
-
-    const ValueType re_value1 = 2000.;
-    const ValueType re_value2 = 4000.;
+    constexpr double re_value1 = 2000.;
+    constexpr double re_value2 = 4000.;
 
     if (re < re_value1) {
-        f = 16. / re;
-    } else if (re > re_value2){
+        // not using the formula directly because of the division with very small w
```

Comment:
For re < re_value1 the flow is laminar and the friction factor should be 64 / re . (Or 16 as in the code here due to division by 4 ), Isn't it better to return a large fixed number for w < epsilon.

Reply:
Now I see what you have done. You have factor out w from the friction factor term to avoid w / w, which is problematic for small w.